### DISCUSSION TITLE
Forgetting pre-trained labels after training ner model with 'en_core_web_sm'

### DISCUSSION THREAD

**Discussion by rammohanrrr** [explosion/spaCy#7666] [Help: Coding & Implementations] [answered]
Hi Team, 
I am trying to train a pre-trained library (sm model).
I can able generated config file and able to train with that config file using CLI.
After training new labels, existing (pre-trained) labels are not identifying.
Can i get suggestions to overcome that problem.

Below is my base_config file.

Expected result after training new labels:
It has to identify pre-trained labels and my own custom labels.

Thanks in advance

```
[paths]
train = null
dev = null

[system]
gpu_allocator = null

[nlp]
lang = "en"
pipeline = ["ner"]
batch_size = 1000

[components]

[components.ner]
source = "en_core_web_sm"
component = "ner"

[components.ner.model]
@architectures = "spacy.TransitionBasedParser.v2"
state_type = "ner"
extra_state_tokens = false
hidden_width = 64
maxout_pieces = 2
use_upper = true
nO = null

[components.ner.model.tok2vec]
@architectures = "spacy.Tok2Vec.v2"

[components.ner.model.tok2vec.embed]
@architectures = "spacy.MultiHashEmbed.v1"
width = ${components.ner.model.tok2vec.encode.width}
attrs = ["ORTH","SHAPE"]
rows = [5000,2500]
include_static_vectors = true

[components.ner.model.tok2vec.encode]
@architectures = "spacy.MaxoutWindowEncoder.v2"
width = 256
depth = 8
window_size = 1
maxout_pieces = 3


[corpora]

[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths.train}
max_length = 2000

[corpora.dev]
@readers = "spacy.Corpus.v1"
path = ${paths.dev}
max_length = 0

[training]
#frozen_components = ["tok2vec", "tagger", "parser", "senter", "attribute_ruler", "lemmatizer"]
dev_corpus = "corpora.dev"
train_corpus = "corpora.train"

[training.optimizer]
@optimizers = "Adam.v1"

[training.batcher]
@batchers = "spacy.batch_by_words.v1"
discard_oversize = false
tolerance = 0.2

[training.batcher.size]
@schedules = "compounding.v1"
start = 100
stop = 1000
compound = 1.001

[initialize]
vectors = "en_core_web_sm"
```

**Comment by polm** [comment, marked as answer]
Forgetting like this is a common problem, called "catastrophic forgetting". You can read more about it and one proposed solution we've had [here](https://explosion.ai/blog/pseudo-rehearsal-catastrophic-forgetting). 

We do have some code for doing the proposed rehearsal in v3, but it's still experimental and doesn't have much documentation yet. If you want to take a deeper look at it you can start [in the docs](https://spacy.io/api/language#resume_training) and dive in to the source. 

If you don't want to use the experimental interface, another option is to make sure your training data has examples of the current labels so that the model doesn't forget. As mentioned in the post above though, you have to be careful that the the model doesn't make simplifying assumptions.
